rclpy/rclpy/time_source.py

```diff
@@ -119,9 +127,19 @@ def _on_parameter_event(self, parameter_list):
                 if parameter.type_ == Parameter.Type.BOOL:
                     self.ros_time_is_active = parameter.value
                 else:
-                    self._node.get_logger().error(
-                        '{} parameter set to something besides a bool'
-                        .format(USE_SIM_TIME_NAME))
+                    node = self._get_node()
+                    if node:
+                        node.get_logger().error(
+                            '{} parameter set to something besides a bool'
+                            .format(USE_SIM_TIME_NAME))
                 break
 
         return SetParametersResult(successful=True)
+
+    def _get_node(self):
+        if self._node_weak_ref is not None:
+            node = self._node_weak_ref()
+            if node is None:
+                raise RuntimeError('Attached node has gone out of scope before expected')
+            return node
+        return None
```

Comment:
I would think this case is worth raising for (trying to do an operation that requires a node, but the time source is not attached to one), and the other case of the weak reference being invalid is expected and not worth raising for.

Reply:
I changed the code to not raise an error in case the weak reference is invalid.
The code previous to this change was working when `self._node` was `None`, so I don't think that an error shold be raised in that case.